spec/models/user_spec.rb

```diff
@@ -63,6 +63,85 @@
     end
   end
 
-  xdescribe '.update_with_phone_number_authentication' do
+  describe "Search" do
+    let!(:user_1) { create(:user, full_name: "Sri Priyanka John") }
+    let!(:user_2) { create(:user, full_name: "Priya Sri Gupta") }
+
+    shared_examples "searches whole words against full_names" do |search_method|
+      ["Sri", "sri", "SRi", "sRi", "SRI", "sRI"].each do |term|
+        it "returns results for case-insensitive searches: #{term.inspect}" do
+          expect(User.send(search_method, term)).to match_array([user_1, user_2])
```

Comment:
Can you change all these `sends` to `public_send` ? `send` should be avoided unless necessary, as it completely bypasses method visibility.